File: src/algorithms/maximum-difference.js
/**
 * Given an array nums of size n, finds the maximum difference between nums[i]
 * and nums[j], such that 0 <= i < j < n and nums[i] < nums[j].
 * 
 * @param {number[]} nums - An array of positive numbers
 * @returns {number} - Returns the max difference
 */
function maximumDifference (nums) {
  let maxDiff = -1;
  for (let i = 0; i < nums.length - 1; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      const diff = nums[j] - nums[i];
      if (diff > maxDiff && diff !== 0) {
        maxDiff = diff;
      }
    }
  }
  return maxDiff;
}

File: src/algorithms/maximum-difference.py
def maximumDifference(nums):
    """Given a 0-indexed integer array nums of size n, find the maximum 
    difference between nums[i] and nums[j] (i.e., nums[j] - nums[i]), such that 
    0 <= i < j < n and nums[i] < nums[j].

    Return the maximum difference. If no such i and j exists, return -1.

    Args:
        nums (list[int]): A list of integers.

    Returns:
        int: Returns the maximum difference between increasing elements or -1.
    """
    res = -1
    for i in range(len(nums) - 1):
        for j in range(i + 1, len(nums)):
            temp = nums[j] - nums[i]
            if temp > res and temp != 0:
                res = temp
    return res
